src/main/java/agh/ics/oop/IWorldMap.java

```diff
@@ -35,7 +38,9 @@ public interface IWorldMap {
      *            Position to check.
      * @return True if the position is occupied.
      */
-    boolean isOccupied(Vector2d position);
+    boolean isOccupiedAnimal(Vector2d position);
+
+    boolean isOccupiedGrass(Vector2d position);
```

Comment:
to zdecydowanie nie powinny być metody interfejsu - nie mają one sensu dla `RectangularMap` (tzw. zasada segregacji interfejsów, możesz poczytać o SOLID generalnie ;) )